crates/trycp_client/src/lib.rs

```diff
@@ -72,10 +72,30 @@ impl TrycpClient {
 
         let (recv_send, recv_recv) = tokio::sync::mpsc::channel(32);
 
+        let ws = Arc::new(tokio::sync::Mutex::new(sink));
+        let ws2 = ws.clone();
         let pend2 = pend.clone();
         let recv_task = tokio::task::spawn(async move {
             while let Some(Ok(msg)) = stream.next().await {
-                let msg = msg.into_data();
+                let msg = match msg {
+                    Message::Close(_) => {
```

Comment:
Is it worth outputting the variant field value?

Reply:
Seems like a good idea, I'll be back here next week adding it if I don't do it now :D